pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -30,8 +30,16 @@ var (
 
 var logger log.Logger = log.New("plugin.instrumentation")
 
-// instrumentPluginRequest instruments success rate and latency of `fn`
-func instrumentPluginRequest(ctx context.Context, cfg *config.Cfg, pluginCtx *backend.PluginContext, endpoint string, fn func() error) error {
+// instrumentPluginRequest instruments success rate and latency of `fn`, and starts a new span
+// with the provided tracer. The span name will be `endpoint`.
+func instrumentPluginRequest(
+	ctx context.Context, cfg *config.Cfg,
+	tracer tracing.Tracer, pluginCtx *backend.PluginContext,
+	endpoint string, fn func() error,
+) error {
+	ctx, span := tracer.Start(ctx, endpoint)
+	defer span.End()
```

Comment:
I would suggest moving this to a tracing client middleware to clean up this function and dependencies to the plugin client